pkg/oci/routes.go

```diff
@@ -232,54 +199,93 @@ func (cp *CloudProvider) ListRoutes(ctx context.Context, clusterName string) ([]
 				continue
 			}
 
-			// TODO CACHE
-			ipDetails, err := cp.client.Networking().GetIPFromOCID(ctx, *r.NetworkEntityId)
-
+			err := cp.mapCidrBlockToNode(ctx, ipToNodeLookup, r, destinationsMap)
 			if err != nil {
-				glog.V(6).Infof("Failed to resolve IP to OCID %v", *r.NetworkEntityId)
-				return nil, errors.WithStack(err)
-			}
-
-			node, foundNode := ipToNodeLookup[*ipDetails.IpAddress]
-
-			target, ok := destinationsMap[*r.CidrBlock]
-
-			glog.V(6).Infof("Evaluate entry: %v %s %s", node.Name, target, *r.CidrBlock)
-
-			if !ok {
-				if !foundNode {
-					destinationsMap[*r.CidrBlock] = ""
-				} else {
-					destinationsMap[*r.CidrBlock] = node.Name
-				}
-			} else {
-				if target != node.Name {
-					glog.V(6).Infof("Route Destination mismatch %s %s %s", target, node.Name, *r.CidrBlock)
-					destinationsMap[*r.CidrBlock] = ""
-				}
+				return nil, err
 			}
-
-			glog.V(6).Infof("Destination Map %s -> %s", *r.CidrBlock, destinationsMap[*r.CidrBlock])
 		}
+
 	}
 
 	var routes []*cloudprovider.Route
-
 	for destinationCidr, target := range destinationsMap {
-		found := true
-
-		if target == "" {
-			found = false
-		}
-
 		routes = append(routes, &cloudprovider.Route{
 			Name:            destinationCidr,
 			TargetNode:      types.NodeName(target),
-			Blackhole:       !found,
+			Blackhole:       !(target == ""),
 			DestinationCIDR: destinationCidr,
 		})
 	}
 
 	glog.V(6).Infof("Listed Routes: %d", len(routes))
 	return routes, nil
 }
+
+func (cp *CloudProvider) mapNodeSubnetToRouteTable(ctx context.Context, node *v1.Node, newRouteTableIds map[string]bool) error {
+	instanceID := util.MapProviderIDToInstanceID(node.Spec.ProviderID)
+
+	vnic, err := cp.client.Compute().GetPrimaryVNICForInstance(ctx, "", instanceID)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	// Should this be defaulted to true generally? Making a change to
+	// 'ListRoutes' seems like an anti-pattern.
+	if *vnic.SkipSourceDestCheck != true {
+		glog.V(6).Info("Set SkipSourceDestCheck on ", vnic)
+		err = cp.client.Networking().UpdateVnic(ctx, *vnic.Id, true)
+		if err != nil {
+			return err
+		}
+	}
+
+	routeTable, ok := subnetIds[*vnic.SubnetId]
+	if !ok {
+		subnet, err := cp.client.Networking().GetSubnet(ctx, *vnic.SubnetId)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		routeTable = *subnet.RouteTableId
+		subnetIds[*vnic.SubnetId] = routeTable
+	}
+
+	glog.V(6).Infof("Found Route Table id %s", routeTable)
+	newRouteTableIds[routeTable] = true
+
+	return nil
+}
+
+func (cp *CloudProvider) mapCidrBlockToNode(ctx context.Context, ipToNodeLookup map[string]*v1.Node,
```

Comment:
For some really subjective feedback, it might be nice to have this function name align with the map that it is updating. i.e. 'updateDestinationsMap', or maybe even better, call the destinationsMap cidrBlockToNodeMap everywhere instead...